cmd/admin-handlers.go

```diff
@@ -242,36 +250,35 @@ func (a adminAPIHandlers) ServerInfoHandler(w http.ResponseWriter, r *http.Reque
 		return
 	}
 
-	// Web service response
-	reply := make([]ServerInfo, len(globalAdminPeers))
-
-	var wg sync.WaitGroup
-
-	// Gather server information for all nodes
-	for i, p := range globalAdminPeers {
-		wg.Add(1)
-
-		// Gather information from a peer in a goroutine
-		go func(idx int, peer adminPeer) {
-			defer wg.Done()
-
-			// Initialize server info at index
-			reply[idx] = ServerInfo{Addr: peer.addr}
+	thisAddr, err := xnet.ParseHost(GetLocalPeer(globalEndpoints))
+	if err != nil {
+		writeErrorResponseJSON(w, toAdminAPIErrCode(ctx, err), r.URL)
+		return
+	}
 
-			serverInfoData, err := peer.cmdRunner.ServerInfo()
-			if err != nil {
-				reqInfo := (&logger.ReqInfo{}).AppendTags("peerAddress", peer.addr)
-				ctx := logger.SetReqInfo(ctx, reqInfo)
-				logger.LogIf(ctx, err)
-				reply[idx].Error = err.Error()
-				return
-			}
+	storage := objectAPI.StorageInfo(ctx)
 
-			reply[idx].Data = &serverInfoData
-		}(i, p)
+	serverInfo := globalNotificationSys.ServerInfo(ctx)
+	serverInfo[*thisAddr] = ServerInfo{
```

Comment:
I am not sure why we are overriding serverInfo contents after calling globalNotificationSys.ServerInfo(ctx), can you add a comment here ?

Reply:
ServerInfo is a map this is just adding local node.